clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -152,8 +152,21 @@ public void setup() {
         this.rebalanceListener = new MockRebalanceListener();
         this.mockOffsetCommitCallback = new MockCommitCallback();
         this.partitionAssignor.clear();
+        this.rebalanceConfig = buildRebalanceConfig(groupInstanceId);
```

Comment:
Related to my other comment below, we change who we initialize `groupInstanceId`, but this seems to result into more manual initialization within the tests than fewer (as someone would expect by the use of a `setup` method). Should we keep the current way of initialization?

Reply:
Sounds good, default to dynamic membership setup